src/ring_theory/algebra.lean

```diff
@@ -26,55 +26,37 @@ ring is the under category R ↓ CRing. In the categorical
 setting we have a forgetful functor R-Alg ⥤ R-Mod.
 However here it extends module in order to preserve
 definitional equality in certain cases. -/
-class algebra (R : Type u) (A : Type v) [comm_ring R] [ring A] extends has_scalar R A :=
-(to_fun : R → A) [hom : is_ring_hom to_fun]
+@[nolint has_inhabited_instance]
```

Comment:
If you want, you can mark the identity as default algebra (i.e. `inhabited` instance).

Reply:
Probably a better choice is `of_ring_hom 0 _` because it works for all `R`, `A`.

Reply:
All ring homs are unital, so that won't work.

Reply:
You're right. That's why I didn't add an `inhabited` instance in the first place. I don't want to add an `inhabited` sequence just to make the linter happy. For me this `nolint` says "there is no universal construction that turns a `semiring` into an algebra over a given `comm_semiring`".

Reply:
Ok, I guess there's not really a consensus on what to do with this linter (-;
I like your point of view, but I also like @semorrison 's idea of using these instances to sprinkle examples into mathlib...